include/LeMonADE/updater/UpdaterAbstractCreate.h

```diff
@@ -0,0 +1,301 @@
+ *
+ * @brief abstract updater to create systems
+ * 
+ * @details This abstract class provides the three basic functions to create systems: add a single monomer, add a connected monomer and move the system to find some free space
+ *
+ * @tparam IngredientsType
+ *
+ **/
+
+#include <LeMonADE/updater/AbstractUpdater.h>
+#include <LeMonADE/utility/MonomerGroup.h>
+#include <LeMonADE/utility/DepthIterator.h>
+#include <LeMonADE/utility/Vector3D.h>
+
+template<class IngredientsType>
+class UpdaterAbstractCreate:public AbstractUpdater
+{
+public:
+  UpdaterAbstractCreate(IngredientsType& ingredients_):ingredients(ingredients_){}
+  
+  virtual void initialize();
+  virtual bool execute();
+  virtual void cleanup();
+  
+protected:
+  IngredientsType& ingredients;
+  RandomNumberGenerators rng;
+  
+  //! function to add a standalone monomer
+  bool add_lonely_monomer(int32_t type=1);
+  
+  //! function to add a monomer to a parent monomer
+  bool add_monomer_to_parent(uint32_t parent_id, int32_t type=1);
+  
+  //! function to add a monomer at a spezific place
+  bool add_monomer_to_position(VectorInt3 pos, int32_t type=1);
+  
+  //! function to add a new monomer inbetween two others
+  bool add_monomer_inside_connected_pair(uint32_t indexA, uint32_t indexB, int32_t type=1);
+  
+  //! function to move the whole system
+  void move_system(int32_t nsteps);
+  
+  //! function to find groups of connected monomers
+  void linearize_system();
+  
+};
+
+/**
+* The initialize function handles the new systems information.
+*
+* @tparam IngredientsType Features used in the system. See Ingredients.
+*/
+template < class IngredientsType >
+void UpdaterAbstractCreate<IngredientsType>::initialize(){
+  
+}
+
+/**
+* Execution of the system creation
+*
+* @tparam IngredientsType Features used in the system. See Ingredients.
+*/
+template < class IngredientsType >
+bool UpdaterAbstractCreate<IngredientsType>::execute(){
+  
+}
+
+/**
+* Standard clean up.
+*
+* @tparam IngredientsType Features used in the system. See Ingredients.
+*/
+template < class IngredientsType >
+void UpdaterAbstractCreate<IngredientsType>::cleanup(){
+  
+}
+
+/******************************************************************************/
+/**
+ * @brief function to add a standalone monomer
+ * @param type attribute tag of the new monomer
+ * @return <b false> if position is not free, <b true> if move was applied
+ */
+template<class IngredientsType>
+bool UpdaterAbstractCreate<IngredientsType>::add_lonely_monomer(int32_t type){
+  // set properties of add Monomer Move
+  MoveAddScMonomer addmove;
+  addmove.init(ingredients);
+  
+  int32_t counter(0);
+  while(counter<10000){
+    VectorInt3 newPosition((rng.r250_rand32() % (ingredients.getBoxX()-1)),
+				  (rng.r250_rand32() % (ingredients.getBoxY()-1)),
+				  (rng.r250_rand32() % (ingredients.getBoxZ()-1)));
+    addmove.setPosition(newPosition);
+    if(addmove.check(ingredients)==true){
+      addmove.setType(type);
```

Comment:
Here and in the other add-functions: Is there any specific thought behind adding the type after the check? I would assume there could exist features, in which the check depends on the type. Thus my suggestion is to have the type set before the check.